src/node_contextify.cc

```diff
@@ -1016,8 +1024,11 @@ void ContextifyContext::CompileFunction(
       Number::New(isolate, loader::ScriptType::kFunction));
   host_defined_options->Set(
       isolate, loader::HostDefinedOptions::kID, Number::New(isolate, id));
-  host_defined_options->Set(
-      isolate, loader::HostDefinedOptions::kFilename, filename);
+  // only set the filename host option for dynamic import support
+  if (dynamic_import) {
+    host_defined_options->Set(
+        isolate, loader::HostDefinedOptions::kFilename, filename);
+  }
 
```

Comment:
☝️ If that `filename` is [like `vm.Script()` option](https://nodejs.org/dist/latest-v11.x/docs/api/vm.html#vm_new_vm_script_code_options) for `filename` then it should not be gated by `dynamic_import`:
> `filename` *\<string\>* Specifies the filename used in stack traces produced by this script.

Reply:
host_defined_options is only for the esm loader, the filename is attributed for debugging in the ScriptOrigin constructor.

Reply:
That filename is still set fine on the next line.